### PR TITLE
Consistency in API response for live brokers

### LINKED ISSUE #11841: Response parity between Table/getLiveBrokers and Table/getLiveBrokersForTable
At present, Table/getLiveBrokers returns response in this form including broker instance, host and port:
API: `/tables/livebrokers` 

```
{
  "tbl1_REALTIME": [
    {
      "instanceName": "Broker_instance1",
      "port": 7000,
      "host": "host1"
    },
    {
      "instanceName": "Broker_instance2",
      "port": 7000,
      "host": "host2"
    }
   ],
"tbl2_REALTIME": [
    {
      "instanceName": "Broker_instance1",
      "port": 7000,
      "host": "host1"
    },
    {
      "instanceName": "Broker_instance2",
      "port": 7000,
      "host": "host2"
    }
   ]
}
```

But tables/{tableName}/livebrokers returns only the list of Broker-Instance-IDs:

```
[
  "Instance1",
  "Instance2"
]
```

Ideally we want this to also return host and port in the response as well.

cc @cypherean who will be working on this.

### PR DESCRIPTION
**Pull request by cypherean**
Resolves #11841

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
This change can be risky because it changes the response format which can break client. A safer option would be to enhance the `/tables/livebrokers` API to take an optional `tableName` as the query parameter. We may annotate `/tables/{tableName}/livebrokers` as deprecated and suggest people to use the new API

**Review by ankitsultana — CHANGES_REQUESTED** [review]
This will break our production from what I understand. @tibrewalpratik17 can you also take a look?

**Comment by tibrewalpratik17** [comment]
> This will break our production from what I understand. @tibrewalpratik17 can you also take a look?

Yes! @Jackie-Jiang's suggestion sounds good to me! 
We can mark this API as deprecated and add an optional param in `/tables/livebrokers`.

In our production, we are making 2 calls per table to get list of broker host:port to query, having a single API to give us broker host:port will reduce one-hop for us.

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableInstances.java [hunk 1]
```diff
@@ -165,9 +166,11 @@ public List<String> getLiveBrokersForTable(
   @ApiResponses(value = {
       @ApiResponse(code = 200, message = "Success"), @ApiResponse(code = 500, message = "Internal server error")
   })
-  public Map<String, List<InstanceInfo>> getLiveBrokers() {
+  public Map<String, List<InstanceInfo>> getLiveBrokers(
+          @ApiParam(value = "Table name (with or without type)", required = false)
+          @QueryParam("tablename") Optional<String> tablename) {
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableInstances.java:171]: Query param should be `tableName`
Shall we keep it consistent with other APIs where optional field is represented as `@Nullable`
```suggestion
          @QueryParam("tableName") @Nullable tableName) {
```

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 2]
```diff
@@ -3923,18 +3935,21 @@ public Map<String, List<InstanceInfo>> getTableToLiveBrokersMapping() {
         .collect(Collectors.toMap(InstanceConfig::getInstanceName, Function.identity()));
 
     Map<String, List<InstanceInfo>> result = new HashMap<>();
+    if (optionalTableName.isPresent()) {
+      String tableName = optionalTableName.get();
+      List<InstanceInfo> hosts = getBrokersFromTableNameWithType(ev, tableName, instanceConfigMap,
+              this::getLiveBrokerInstancesForTable);
+      if (!hosts.isEmpty()) {
+        result.put(tableName, hosts);
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java:3943]: I think we want to put hosts when table exists, even if it is empty

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 3]
```diff
@@ -3923,18 +3935,21 @@ public Map<String, List<InstanceInfo>> getTableToLiveBrokersMapping() {
         .collect(Collectors.toMap(InstanceConfig::getInstanceName, Function.identity()));
 
     Map<String, List<InstanceInfo>> result = new HashMap<>();
+    if (optionalTableName.isPresent()) {
+      String tableName = optionalTableName.get();
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java:3939]: This `tableName` can be raw (without type suffix). The returned map should contain the actual table name with type suffix

**Comment by cypherean** [comment]
Closing this to address all comments in a new PR and opening #12201

**Pull request closed without merging**